docs/src/proposals/stake-v2.md

```diff
@@ -0,0 +1,211 @@
+at epoch boundaries automatically.
+
+## New Stake Program Instructions
+
+`Split` / `Merge` / `Deactivate` are only allowed if the stake account has
+been updated this epoch, or after deactivation epoch if applicable.
+
+Side note: deactivate might actually be OK, since we could technically calculate
+the rewards even after deactivation.
+
+### UpgradeToV2
+
+Given a V1 stake account and a signature from the stake or withdraw authority,
+reallocate the existing stake account and convert it to a V2 stake. Also reduces
+the delegation due to the higher rent-exemption. Fails if the stake account
+doesn't have enough lamports to cover the new rent-exemption.
+
+Side note: this would also be a great time to include the rent-exemption in the
+delegation. That way, smaller stakers get a tiny bit more, and there's a couple of other
+nice side effects. Ask me about it.
+
+### Redelegate
+
+Given a V2 stake account, vote account, and stake authority signature, change the
+redelegation. Follows most of the same rules as delegate.
+
+If it's going through a multi-epoch redelegation, be sure that previous epoch
+rewards have been claimed, and update the vote pubkey / credits / epoch.
+
+### (Also in the Vote Program) ClaimRewards
+
+Given only a V2 stake account and the delegated vote account (or two vote accounts
+during multi-epoch redelegation), calculate the rewards gained since the last
+claim, and withdraw the lamports from the vote account.
+
+For easier use with existing tools, if a redelegation is complete, move the
+redelegation information over into the main delegation.
+
+Implementation note: claiming lamports from the vote account and also updating the
+stake account is tricky since only the owning program can deduct lamports or modify
+data, and we need to do both conditionally.
+
+Potential solution: the vote program CPIs into the stake program to update the
+credits observed, then transfers the lamports from the vote account to the stake account.
+The stake program must enforce that only the vote program can call it, otherwise
+it's possible to deprive an account of their rewards.
+
+BIG ISSUE TO CONSIDER: vote accounts only store 64 epochs of credits, which means
+that we can only accurately calculate 64 epochs of credits.
+
+Potential solution: if it's been too long, drop rewards. This is easiest, but not great.
+We can include a mechanism where anyone can claim a tiny portion of rewards from
+an account if the update is done during the 64th epoch since the last claim.
+
+## New Vote Account Field
+
+Vote accounts need to keep track of `withheld_rewards`. We should be able to steal
+some bytes from `prior_voters`, since we only need a `u64` for the withheld amount.
+
+`Withdraw` needs to respect the withheld amount and rent-exemption. This means
+that vote accounts cannot be deleted if any withheld rewards remain. Thankfully,
+since claiming rewards is permissionless, a node operator can do this themselves
```

Comment:
> claiming rewards is permissionless

TIL.  probably should mention this in the `ClaimRewards` section.  It's kinda alluded to in the aside about handling vote history expiration, but it sounds like a firm commitment here